### PR TITLE
refactor(): Viewer update

### PR DESCRIPTION
**Pull request by kaosat-dev**
WORK IN PROGRESS DO NOT MERGE
This merges in the 'csg viewer', simplifies it, and merges it with the new renderer

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [x] Does your submission pass tests?

> Note: please do NOT include build files (those generate by the build-xxx commands) with your PR,

Thank you for your help in advance, much appreciated !

### CODE REVIEW

File: packages/ui/csg-viewer/package.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    "dev": "budo example/demo.js --title csg-viewer --live --css example/demo.css -- -t glslify -t brfs",
+    "build": "browserify -p common-shakeify --standalone makeCsgViewer src/index.js > dist/bundle.js",
+    "minify": "uglifyjs dist/bundle.js --compress > dist/bundle.min.js",
+    "compress": "gzipme dist/bundle.min.js",
+    "build-min": "npm run build && npm run minify && npm run compress"
+  },
+  "author": "Mark 'kaosat-dev' Moissette",
+  "license": "MIT",
+  "devDependencies": {
+    "@jscad/scad-api": "^0.4.2",
```
Comment (z3dev) [inline packages/ui/csg-viewer/package.json]: Why the dependency on @jscad/scad-api?

Comment (kaosat-dev) [reply packages/ui/csg-viewer/package.json]: devDependency: ie for tests/ demo to illustrate how you can use to view your csg stuff
we can remove that in the course of V2

**Comment by z3dev** [comment]
This is firmly large change. Do you want a review? Should I test these changes?

**Comment by z3dev** [comment]
Feedback...
- no issues with installation, bootstrap, or running dev
- the smoothness of the viewer is very nice; rotation, zoom, panning
- for some reason, dark theme doesn't work. i would really like this for the show next weekend (8/3)

I think this is a good base for making further changes. V2 is on the way! Awesome!

@kaosat-dev please merge if you are satisfied with this integration.

**Comment by kaosat-dev** [comment]
Thanks for the feedback @z3dev !
Going to merge, and will try to fix dark theme in another PR (was broken a while back, further down the line we will be able to add unit tests for UI ;)

**Pull request merged**